fe/fe-core/src/main/java/org/apache/doris/blockrule/SqlBlockRuleMgr.java

```diff
@@ -0,0 +1,222 @@
+            SqlBlockRule originRule = nameToSqlBlockRuleMap.get(ruleName);
+            if (StringUtils.isEmpty(sqlBlockRule.getSql())) {
+                sqlBlockRule.setSql(originRule.getSql());
+            }
+            if (StringUtils.isEmpty(sqlBlockRule.getSqlHash())) {
+                sqlBlockRule.setSqlHash(originRule.getSqlHash());
+            }
+            if (sqlBlockRule.getGlobal() == null) {
+                sqlBlockRule.setGlobal(originRule.getGlobal());
+            }
+            if (sqlBlockRule.getEnable() == null) {
+                sqlBlockRule.setEnable(originRule.getEnable());
+            }
+            unprotectedUpdate(sqlBlockRule);
+            Catalog.getCurrentCatalog().getEditLog().logAlterSqlBlockRule(sqlBlockRule);
+        } finally {
+            writeUnlock();
+        }
+    }
+
+    public void replayAlter(SqlBlockRule sqlBlockRule) {
+        unprotectedUpdate(sqlBlockRule);
+        LOG.info("replay alter sql block rule: {}", sqlBlockRule);
+    }
+
+    public void unprotectedUpdate(SqlBlockRule sqlBlockRule) {
+        nameToSqlBlockRuleMap.put(sqlBlockRule.getName(), sqlBlockRule);
+    }
+
+    public void unprotectedAdd(SqlBlockRule sqlBlockRule) {
+        nameToSqlBlockRuleMap.put(sqlBlockRule.getName(), sqlBlockRule);
+        String sql = sqlBlockRule.getSql();
+        if (StringUtils.isNotEmpty(sql)) {
+            sqlPatternMap.put(sql, Pattern.compile(sql));
+        }
+    }
+
+    public void dropSqlBlockRule(DropSqlBlockRuleStmt stmt) throws DdlException {
+        writeLock();
+        try {
+            List<String> ruleNames = stmt.getRuleNames();
+            for (String ruleName : ruleNames) {
+                if (!existRule(ruleName)) {
+                    throw new DdlException("the sql block rule " + ruleName + " not exist");
+                }
+            }
+            unprotectedDrop(ruleNames);
+            Catalog.getCurrentCatalog().getEditLog().logDropSqlBlockRule(ruleNames);
+        } finally {
+            writeUnlock();
+        }
+    }
+
+    public void replayDrop(List<String> ruleNames) {
+        unprotectedDrop(ruleNames);
+        LOG.info("replay drop sql block ruleNames: {}", ruleNames);
+    }
+
+    public void unprotectedDrop(List<String> ruleNames) {
+        ruleNames.forEach(name -> nameToSqlBlockRuleMap.remove(name));
+    }
+
+    public void matchSql(String originSql, String sqlHash, String user) throws AnalysisException {
+        // match global rule
+        List<SqlBlockRule> globalRules = nameToSqlBlockRuleMap.values().stream().filter(SqlBlockRule::getGlobal).collect(Collectors.toList());
+        for (SqlBlockRule rule : globalRules) {
+            matchSql(rule, originSql, sqlHash);
+        }
+        // match user rule
+        String bindSqlBlockRules = Catalog.getCurrentCatalog().getAuth().getSqlBlockRules(user);
+        if (StringUtils.isNotEmpty(bindSqlBlockRules)) {
+            String[] split = bindSqlBlockRules.split(",");
```

Comment:
This `split()` should not be done every time we call `matchSql`.
It should be prepared already.